### DISCUSSION TITLE
Query about Digital Lowpass filter in the mpu6050

### DISCUSSION THREAD

**Discussion by AsifKhan991** [jrowberg/i2cdevlib#626] [Q&A]
Hi,
I want to get the real body acceleration of the three axis without gravity component with DLPF set to 5Hz rate. I am using Arduino platform. How can I achieve that and which files to edit?

**Comment by ZHomeSlice** [comment]
@AsifKhan991 
Take a look at: [https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Register-Map1.pdf](https://invensense.tdk.com/wp-content/uploads/2015/02/MPU-6000-Register-Map1.pdf)
Page 13

The default value for this register Is 1: 
`	I2Cdev::writeBytes(devAddr,0x1A, 1, &(val = 0x01), wireObj); // 0000 0001 CONFIG: Digital Low Pass Filter (DLPF) Configuration 188HZ  //Im betting this will be the beat
`
you can directly update the register 
I believe you will want to change it to 6: 
`	I2Cdev::writeBytes(devAddr,0x1A, 1, &(val = B00000110), wireObj); // 0000 0110 CONFIG: Digital Low Pass Filter (DLPF) Configuration 5HZ  
`

or
you could use this function if you would like

mpu.setDLPFMode(6);
mpu.setDLPFMode(B00000110);

source for the function:
`MPU6050_RA_CONFIG is set to 1A`

```

/** Set digital low-pass filter configuration.
 * @param mode New DLFP configuration setting
 * @see getDLPFBandwidth()
 * @see MPU6050_DLPF_BW_256
 * @see MPU6050_RA_CONFIG
 * @see MPU6050_CFG_DLPF_CFG_BIT
 * @see MPU6050_CFG_DLPF_CFG_LENGTH
 */
void MPU6050_Base::setDLPFMode(uint8_t mode) {
    I2Cdev::writeBits(devAddr, MPU6050_RA_CONFIG, MPU6050_CFG_DLPF_CFG_BIT, MPU6050_CFG_DLPF_CFG_LENGTH, mode, wireObj);
}
```

Z
